docker-compose.yml

```diff
@@ -0,0 +1,211 @@
+version: '3.8'
+
+services:
+  angular-build:
+    build:
+      context: .
+      dockerfile: ./docker/Dockerfile
+    container_name: oppia-angular-build
+    volumes:
+      - .:/app/oppia
+      - shared_node_modules:/app/oppia/node_modules
+      - shared_third_party:/app/oppia/third_party
+      - shared_proto_files:/app/oppia/proto_files
+      - shared_js_ts_proto_files:/app/oppia/extensions/classifiers/proto
+      - shared_assets_dir:/app/oppia/assets
```

Comment:
When do we create this?

Reply:
within build.py, some hashes file is generated. if not mount this vol, then there were error in the logs that `/assets/hashes.json` not found

Reply:
When do we generate this file? Like, why is this not generated now i.e,  why do we have to mount this instead of generating it? (I'm fine addressing this comment later once we have all the services ready!)

Reply:
> When do we generate this file?
the file is generated with the build.py execution
>  why do we have to mount this instead of generating it
it is not present in the local dir, generated with the build.py script. so we need to mount all the generated files so that it can be used by the required services